src/NServiceBus.Core/Recoverability/RecoverabilitySettings.cs

```diff
@@ -0,0 +1,236 @@
+        /// </summary>
+        public TimeSpan TimeIncrease { get; }
+    }
+
+    /// <summary>
+    /// Provides information about the fault configuration.
+    /// </summary>
+    public struct FailedConfig
+    {
+        internal FailedConfig(string errorQueue)
+        {
+            ErrorQueue = errorQueue;
+        }
+
+        /// <summary>
+        /// Gets the configured standard error queue.
+        /// </summary>
+        public string ErrorQueue { get; }
+    }
+
+    /// <summary>
+    /// Extension methods for recoverability which extend <see cref="EndpointConfiguration" />.
+    /// </summary>
+    public static class RecoverabilityEndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Configuration settings for recoverability.
+        /// </summary>
+        /// <param name="configuration">The endpoint configuration.</param>
+        public static RecoverabilitySettings Recoverability(this EndpointConfiguration configuration)
+        {
+            return new RecoverabilitySettings(configuration.GetSettings());
+        }
+    }
+
+    /// <summary>
+    /// Configuration settings for immediate retries.
+    /// </summary>
+    public class ImmediateRetriesSettings : ExposeSettings
+    {
+        internal ImmediateRetriesSettings(SettingsHolder settings) : base(settings)
+        {
+        }
+
+        /// <summary>
+        /// Configures the amount of times a message should be immediately retried after failing before escalating to second level
+        /// retries.
+        /// </summary>
+        /// <param name="numberOfRetries">The number of times to immediately retry a failed message.</param>
+        public void NumberOfRetries(int numberOfRetries)
+        {
+            Guard.AgainstNegative(nameof(numberOfRetries), numberOfRetries);
+
+            Settings.Set(Recoverability.FlrNumberOfRetries, numberOfRetries);
+        }
+    }
+
+    /// <summary>
+    /// Configuration settings for delayed retries.
+    /// </summary>
+    public class DelayedRetriesSettings : ExposeSettings
+    {
+        internal DelayedRetriesSettings(SettingsHolder settings) : base(settings)
+        {
+        }
+
+        /// <summary>
+        /// Configures the number of times a message should be retried with a delay after failing first level retries.
+        /// </summary>
+        public DelayedRetriesSettings NumberOfRetries(int numberOfRetries)
+        {
+            Guard.AgainstNegative(nameof(numberOfRetries), numberOfRetries);
+
+            Settings.Set(Recoverability.SlrNumberOfRetries, numberOfRetries);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures the delay interval increase for each failed second level retry attempt.
+        /// </summary>
+        public DelayedRetriesSettings TimeIncrease(TimeSpan timeIncrease)
+        {
+            Guard.AgainstNegative(nameof(timeIncrease), timeIncrease);
```

Comment:
didn't simon add some more advanced checks in the old code to make sure that it's within a reasonable timespan?